### PR TITLE
Add parser and dataset demonstration

### PR DESCRIPTION
**Pull request by VyrodovMikhail**

### CODE REVIEW

**Comment by shmpanski** [comment]
Hi, Mikhail! Thanks for this cool PR, you're good at writing code!

There are few points and issues:
1. It's better to use english for open source projects
2. I think you should refactor your repo. In case of using scrapy, this framework generates python module inside your base directory. It forces you not to use the `src` directory, but it's okay. Try to follow the next structure:
```
├── LICENSE
├── README.md
├── goat
│   ├── __init__.py
│   ├── dataset_demonstration.py
│   ├── dataset_utils.py
│   ├── items.py
│   ├── middlewares.py
│   ├── pipelines.py
│   ├── settings.py
│   ├── spider_utils.py
│   └── spiders
│       ├── __init__.py
│       └── ege_spider.py
├── pyproject.toml
├── requirements.dev.txt
├── requirements.txt
├── scrapy.cfg
├── scripts
│   └── train.py
└── tests
    └── test_model.py
```
@SpirinEgor, review it please. Scrapy wants to have `flat` layout of project, not `src` (as I understand, this lock is required for running `scrapy crawl` command.

3. Don't forget to update `pyproject.toml` with actual project name and info

**Comment by SpirinEgor** [comment]
It's okay for me to change template, don't forget to update CI config too

File: goat/dataset_demonstration.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import sys
+
+import pandas as pd
+from dataset_utils import remove_unnecessary_symbols
+from spider_utils import TaskType
+
+file_path = f"./{sys.argv[1]}"
```
Comment (shmpanski) [inline goat/dataset_demonstration.py:7]: You can use [argparse](https://docs.python.org/3/library/argparse.html) lib for parsing input arguments.

If you are writing a script in python, it is better to put its logic in a separate method and call it inside the if-name-main construct:
```python
if __name__ == "__main__":
    args = ...
    my_script_function(*args)
```

File: goat/items.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    solution_images = scrapy.Field()
+    answer = scrapy.Field()
+    url = scrapy.Field()
+    fipi_links = scrapy.Field()
+    sources_links = scrapy.Field()
+    task_points = scrapy.Field()
+    criteria_table = scrapy.Field()
+    html_task = scrapy.Field()
+
+    pass
```
Comment (shmpanski) [inline goat/items.py:28]: It seems that this "pass" is not necessary

File: goat/spider_utils.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    "DE": f"https://de.{_RESHU_CT}",
+    "FR": f"https://fr.{_RESHU_CT}",
+    "LIT": f"https://lit.{_RESHU_CT}",
+    "SP": f"https://sp.{_RESHU_CT}",
+    "WH": f"https://wh.{_RESHU_CT}",
+    "BH": f"https://bh.{_RESHU_CT}",
+}
+
+
+def determine_soc_task_type(exam_type: str, topic_id: str) -> tuple[TaskType, bool]:
```
Comment (shmpanski) [inline goat/spider_utils.py:94]: You can do a little refactoring here and pull all these constants into a separate mapping and check the type inside the loop. This is not necessary, it's a matter of code beautification.

**Comment by shmpanski** [comment]
@VyrodovMikhail, you can also check the errors that mypy gives during CI pipeline.
@SpirinEgor, maybe we should consider turning off mypy checks.

File: goat/dataset_demonstration.py (hunk 1 above)
Comment (VyrodovMikhail) [reply goat/dataset_demonstration.py]: Done

File: goat/items.py (hunk 2 above)
Comment (VyrodovMikhail) [reply goat/items.py]: Done

**Review by shmpanski — APPROVED** [review]

**Pull request merged**